logd/src/logd/rpc_forwarder.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "exceptions.h"
+#include "proto_converter.h"
+#include "rpc_forwarder.h"
+#include <vespa/log/exceptions.h>
+#include <vespa/vespalib/util/buffer.h>
+#include <vespa/vespalib/util/stringfmt.h>
+
+#include <vespa/log/log.h>
+LOG_SETUP(".logd.rpc_forwarder");
+
+using ns_log::BadLogLineException;
+using ns_log::LogMessage;
+using vespalib::make_string;
+
+namespace logdemon {
+
+RpcForwarder::RpcForwarder(const vespalib::string &hostname, int rpc_port,
+                           double rpc_timeout_secs, size_t max_messages_per_request)
+    : _connection_spec(make_string("tcp/%s:%d", hostname.c_str(), rpc_port)),
+      _rpc_timeout_secs(rpc_timeout_secs),
+      _max_messages_per_request(max_messages_per_request),
+      _supervisor(),
```

Comment:
consider keeping the supervisor external to enable re-connect without having to re-create 'the world'